modules/opensearch/opensearch.client.inc

```diff
@@ -441,66 +441,73 @@ function opensearch_execute(TingClientRequest $request) {
  *   The collections found in TingClientSearchResult object.
  * @param array $opensearch_static_fast
  *   The fast static cache used by opensearch_execute().
+ * @param TingClientRequest $request
+ *   The request used to get the collections.
  *
  * @throws \TingClientException
  *   If the search client request fails.
  */
-function _opensearch_predict_cache(array $collections, array &$opensearch_static_fast) {
+function _opensearch_predict_cache(array $collections, array &$opensearch_static_fast, TingClientRequest $request) {
```

Comment:
Instead of this change I would much prefer that `_opensearch_predict_cache()` took a [`TingSearchResultInterface`](https://github.com/ding2/ding2/blob/c668d5e52c165b62efc55a4efde7796a5e3c4943/modules/ting/src/Search/TingSearchResultInterface.php#L14) object as the first argument. You can retrieve both resulting collections and the original request from the response.

Reply:
From opensearch_execute() which uses this function I dont have the OpenSearchTingSearchResult which is the only object that implements that interface?

OpenSearchTingSearchResult is first created in opensearch_search_search(), which calls opensearch_do_search() which then calls opensearch_execute().

Reply:
You are absolutely right. Sorry about the confusion.